pkg/controller/accountpool/accountpool_controller.go

```diff
@@ -155,15 +139,54 @@ func (r *ReconcileAccountPool) Reconcile(request reconcile.Request) (reconcile.R
 	return reconcile.Result{}, nil
 }
 
-func updateAccountPoolStatus(currentAccountPool *awsv1alpha1.AccountPool, unclaimedAccounts int, claimedAccounts int) bool {
-	if currentAccountPool.Status.PoolSize != currentAccountPool.Spec.PoolSize {
-		return true
-	} else if currentAccountPool.Status.UnclaimedAccounts != unclaimedAccounts {
-		return true
-	} else if currentAccountPool.Status.ClaimedAccounts != claimedAccounts {
-		return true
-	} else {
-		return false
+// Calculates the unclaimedAccountCount and Claimed Account Counts
+func (r *ReconcileAccountPool) calculateAccountStatus() (awsv1alpha1.AccountPoolStatus, error) {
+	unclaimedAccountCount := 0
+	claimedAccountCount := 0
+	availableAccounts := 0
+	accountsProgressing := 0
+
+	//Get the number of actual unclaimed AWS accounts in the pool
+	accountList := &awsv1alpha1.AccountList{}
+
+	listOpts := []client.ListOption{
+		client.InNamespace(awsv1alpha1.AccountCrNamespace),
+	}
+	if err := r.client.List(context.TODO(), accountList, listOpts...); err != nil {
+		return awsv1alpha1.AccountPoolStatus{}, err
 	}
 
+	for _, account := range accountList.Items {
+		// count unclaimed vs claimed accounts
+		// We don't want to count reused accounts here, filter by LegalEntity.ID
+		if account.HasNeverBeenClaimed() {
+			if !account.IsFailed() {
+				unclaimedAccountCount++
+			}
+		} else {
+			claimedAccountCount++
+		}
+
+		// count available accounts
+		if account.HasNeverBeenClaimed() && account.IsReady() {
```

Comment:
Could be moved into the if block above

Reply:
Yeah, I was kind of torn on doing that, or keeping it explicitly separate to keep the logic easier to follow if it changes more in the future.

I was even thinking about getting rid of the "else" block above as well, and just keeping each individual calculation separate just for an easier glance as to what's doing what instead of overloading the if statements and making it harder to follow.

Thoughts?

Reply:
Removing the else might actually make it more difficult to read, `if !account.HasNeverBeenClaimed() {` has not never been claimed == has been claimed before, translation has to happen in your head. 

Personally I don't see a problem with moving this block up to into other once it's paired with a comment to clarify any potential confusion - but I'm happy with either solution tbh :)

Reply:
I ended up just creating a simpler method `HasBeenClaimedAtLeastOnce()` which describes this a bit better.